packages/auth-construct/src/construct.test.ts

```diff
@@ -1480,6 +1481,34 @@ void describe('Auth construct', () => {
       const outputs = template.findOutputs('*');
       assert.equal(outputs['socialProviders']['Value'], `["GOOGLE"]`);
     });
+    void it('can override group precedence and correctly updates stored output', () => {
+      const app = new App();
+      const stack = new Stack(app);
+      const auth = new AmplifyAuth(stack, 'test', {
+        loginWith: { email: true },
+        groups: ['admins', 'managers'],
+      });
+      auth.resources.groups['admins'].cfnUserGroup.precedence = 2;
```

Comment:
What happens if this was overridden to `1`, i.e. conflict with the other group's precedence?

Reply:
For us this is possible and Cognito allows it, only issue is in the frontend when they get credentials, the `preferred_role` is not set if the role ARN is different. See https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-user-groups.html#assigning-precedence-values-to-groups.